### PR TITLE
feat: add base64 function to encode binary data

### PR DESCRIPTION
**Pull request by eugenels**

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:disappointed: **fail** : 10 / 33 (30.30%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bin/Base64FunctionFactory.java|10|33|30.30%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:disappointed: **fail** : 10 / 33 (30.30%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bin/Base64FunctionFactory.java|10|33|30.30%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:disappointed: **fail** : 10 / 33 (30.30%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bin/Base64FunctionFactory.java|10|33|30.30%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 25 / 26 (96.15%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bin/Base64FunctionFactory.java|5|6|83.33%|
|:large_blue_circle:|io/questdb/std/Chars.java|20|20|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 25 / 26 (96.15%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bin/Base64FunctionFactory.java|5|6|83.33%|
|:large_blue_circle:|io/questdb/std/Chars.java|20|20|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 28 / 33 (84.85%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bin/Base64FunctionFactory.java|8|13|61.54%|
|:large_blue_circle:|io/questdb/std/Chars.java|20|20|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 28 / 33 (84.85%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bin/Base64FunctionFactory.java|8|13|61.54%|
|:large_blue_circle:|io/questdb/std/Chars.java|20|20|100.00%|

File: core/src/main/java/io/questdb/griffin/engine/functions/bin/Base64FunctionFactory.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    public Function newInstance(int position,
+                                ObjList<Function> args,
+                                IntList argPositions,
+                                CairoConfiguration configuration,
+                                SqlExecutionContext sqlExecutionContext) throws SqlException {
+
+        final int maxLength = args.get(1).getInt(null);
+
+        if (maxLength < 1) {
+            throw SqlException.$(argPositions.getQuick(1), "maxLength has to be greater than 0");
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/engine/functions/bin/Base64FunctionFactory.java:56]: could we also limit max length from top, perhaps to configuration value to make sure system cannot be abused ?

File: core/src/main/java/io/questdb/griffin/engine/functions/bin/Base64FunctionFactory.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        }
+
+        Function func = args.get(0);
+        if (func.isConstant()) {
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/engine/functions/bin/Base64FunctionFactory.java:60]: this is really not necessary, SQL optimiser does this for you

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 28 / 33 (84.85%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bin/Base64FunctionFactory.java|8|13|61.54%|
|:large_blue_circle:|io/questdb/std/Chars.java|20|20|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 30 / 30 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/Chars.java|20|20|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bin/Base64FunctionFactory.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 30 / 30 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/Chars.java|20|20|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bin/Base64FunctionFactory.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|1|1|100.00%|

**Review by bluestreak01 — APPROVED** [review]

**Pull request merged**